rpc/grpc/client_server.go

```diff
@@ -34,6 +55,23 @@ func StartGRPCClient(protoAddr string) BroadcastAPIClient {
 	return NewBroadcastAPIClient(conn)
 }
 
+// StartBlockAPIGRPCClient dials the gRPC server using protoAddr and returns a new
+// BlockAPIClient.
+func StartBlockAPIGRPCClient(protoAddr string, opts ...grpc.DialOption) (BlockAPIClient, error) {
+	if len(opts) == 0 {
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	}
+	opts = append(opts, grpc.WithContextDialer(dialerFunc))
+	conn, err := grpc.Dial( //nolint:staticcheck
```

Comment:
This can be refactored to eliminate the nolint directive

```suggestion
	conn, err := grpc.NewClient(
```

Reply:
The problem with this is that it uses an address from the config which has a tcp prefix and when used with NewClient it fails to connect.

So, changing this would mean reparsing the provided address and separating it to `host:port`. 

We can create an issue to do it along with the other gRPC endpoint for consistency instead of touching both in here

Reply:
https://github.com/celestiaorg/celestia-core/issues/1587